pages/en/lb4/Server.md

```diff
@@ -0,0 +1,220 @@
+  constructor(
+    @inject(CoreBindings.APPLICATION_INSTANCE) app: Application,
+    @inject(MyServerBindings.CONFIG) options?: MyServerConfig
+  ) {
+    super(app);
+  }  
+}
+
+export interface MyServerConfig {
+  x: string;
+  y: number;
+}
+```
+
+### Implement Server Functions
+
+You may have already noticed that the Server declaration above also implements a `Server` interface.
+There are two basic functions a Server must have:
+
+- start()
+- stop()
+
+`start` spins up the server to a working state, keep doing one or more things 
+and could interact with events from outside.
+
+`stop` stops a server by exiting its working state.
+
+{% include code-caption.html content="myServer/src/MyServer.ts" %}
+```js
+import {Server, Application, CoreBindings} from '@loopback/core';
+import {Context, inject} from '@loopback/context';
+import * as EventEmitter from 'events';
+
+class MyEmitter extends EventEmitter {}
+
+export class MyServer extends Context implements Server {
+  myEmitter: MyEmitter;
+
+  constructor(
+      @inject(CoreBindings.APPLICATION_INSTANCE) public app: Application
+  ) {
+    super(app);
+    // an Emitter listen on a certain event whose name is returned by `getEventName()`
+    this.myEmitter = new MyEmitter();
+  }
+
+  // return the event name
+  async getEventName(): Promise<string> {
+    return await this.get('MyServerBindings.EMITTER_EVENT');
+  }
+
+  async start(): Promise<void> {
+    console.log('Your server startted!');
+    let eventName = await this.getEventName();
+    // listen on an event and give some response
+    this.myEmitter.on(eventName, () => {
+      console.log('Your server is responding to an event...');
+    });
+    return Promise.resolve();
+  }
+
+  async stop(): Promise<void> {
+    let eventName = await this.getEventName();
+    // remove listeners
+    this.myEmitter.removeAllListeners(eventName);
+    console.log('Your server stopped!');
+    return Promise.resolve();
+  }
+}
+```
+
+### Bind a Server to Application
+
+Now you can bind a Server to your Application by calling `this.server()` 
+in the Application constructor.
+
+{% include code-caption.html content="application.ts" %}
+```js
+import { Application, ApplicationConfig } from '@loopback/core';
+import {MyServer} from './myServer/src/MyServer';
+
+export class MyApp extends Application {
+  constructor(options?: ApplicationConfig) {
+  super(options);
+  this.server(MyServer, 'serverFoo');
+  }
+}
+```
+
+When instanciating `MyApp`, a server instance named `serverFoo` will be 
+bound to the app, `start()` and `stop()` will be mounted to `Application` level functions.
+
+There are several ways to bind a server to an app, the strategy in the example above
+is the most staightforward one:
+
+```js
+app.server(MyServer, 'nameOfYourServer');
+```
+
+Or you can levarage a component to do it:
+
+```js
+const app = new MyApplication({
+  components: [MyComponent]
+});
+// OR
+app.component(MyComponent);
```

Comment:
Components _can_ setup servers, but that's only because components can do literally _anything_ the developer writes them to do!

I would avoid mentioning this, especially given that there's a good chance we'll remove the automatic instantiation of a RestServer from RestComponent (and possibly do away with RestComponent entirely!)

Hidden instantiation of servers behind components isn't necessarily an anti-pattern, but it doesn't do enough to let you see what's happening and could lead to unforeseen consequences.

Reply:
ok